snapcraft/meta/snap_yaml.py

```diff
@@ -77,14 +81,83 @@ class SnapApp(YamlModel):
     command_chain: Optional[List[str]]
     sockets: Optional[Dict[str, Socket]]
 
-    class Config:  # pylint: disable=too-few-public-methods
-        """Pydantic model configuration."""
 
-        allow_population_by_field_name = True
-        alias_generator = lambda s: s.replace("_", "-")  # noqa: E731
+class ContentPlug(_SnapMetadataModel):
+    """Content plug definition in the snap metadata."""
+
+    interface: Literal["content"]
+    target: str
+    content: Optional[str]
+    default_provider: Optional[str]
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]) -> "ContentPlug":
+        """Create and populate a new ``ContentPlug`` object from dictionary data.
+
+        :param data: The dictionary data to unmarshal.
+        :return: The newly created object.
+        :raise TypeError: If data is not a dictionary.
+        """
+        if not isinstance(data, dict):
+            raise TypeError("data is not a dictionary")
+
+        return cls(**data)
+
+    @validator("target")
+    @classmethod
+    def _validate_target_not_empty(cls, val):
+        if val == "":
+            raise ValueError("value cannot be empty")
+        return val
+
+    @property
+    def provider(self) -> Optional[str]:
+        """Return the default content provider name."""
+        if self.default_provider is None:
+            return None
+
+        if ":" in self.default_provider:
+            return self.default_provider.split(":")[0]
```

Comment:
pedantic: add a comment about why we grab the data before the colon

Reply:
Will add comments and debug messages as part of the next PR.